### ISSUE TITLE
smp::submit_to performance problems

### ISSUE THREAD

**Issue by xemul** [scylladb/seastar#1017]
As shown in 0c2228a4916dc2dc66e9d587343ce436b92b7bef, on a busy target shard the `smp::submit_to` can take ~50ms to complete. Partially this is because a task that arrives from a smp message queue is put at the tail of the scheduler queue which is full of busyloops. However, in the example from the mentioned commit message the maximum time to traverse the queue should be `nr_tasks * task_latency = 100 * 50us =  5ms`.

**Comment by avikivity** [comment]
Can't it take an arbitrary amount of time? If the task group in question is busy, it will be delayed.

**Comment by xemul** [comment]
Well, the test in question uses the single task group -- the default one -- and it _is_ constantly busy. But my point is that since the arrived smp job is queued at the tail of the task group's queue, reactor should get to it once it scans this list at least once, i.e. in 5ms.

**Comment by xemul** [comment]
Well, the test in question uses the single task group -- the default one -- and it _is_ constantly busy. But my point is that since the arrived smp job is queued at the tail of the task group's queue, reactor should get to it once it scans this list at least once, i.e. in 5ms.

**Comment by avikivity** [comment]
That's true for non-submit_to tasks as well, no? We have FIFO order usually.

**Comment by xemul** [comment]
Yes, but 5ms to process the task list should result in 1s/5ms=200 messages/s, but the result is 20 messages/s

**Comment by avikivity** [comment]
What's the relationship to this issue?
